### PR TITLE
Analytics: refactor Plugins page view tracking

### PR DESCRIPTION
**Pull request by vindl**
Replace direct `analytics.pageView.record` calls with `PageViewTracker` in `/plugins`.

Check out the updated recommendations on page view tracking:
https://github.com/Automattic/wp-calypso/blob/master/client/lib/analytics/docs/page-views.md

### Testing instructions

1. Set `localStorage.setItem( 'debug', 'calypso:analytics:tracks' )` in the browser console to enable the debug.
2. Routes to check:
  - `/plugins`
  - `/plugins/:plugin`
  - `/plugins/:site`
  - `/plugins/:plugin/:site`
  - `plugins/manage`
  - `plugins/manage/:site`
  - `/plugins/new/:site`
  - `/plugins/popular/:site`
  - `/plugins/featured/:site`
  - `/plugins/setup/:site` (Jetpack only)

### CODE REVIEW

**Comment by gwwar** [comment]
Tested
- [ ] /plugins
- [ ] /plugins/:plugin
- [ ] /plugins/:site
- [ ] /plugins/:plugin/:site
- [ ] plugins/manage
- [ ] plugins/manage/:site
- [ ] /plugins/new/:site
- [ ] /plugins/popular/:site
- [ ] /plugins/featured/:site
- [ ] /plugins/setup/:site (Jetpack only)

File: client/my-sites/plugins/jetpack-plugins-setup/index.jsx [hunk 1]
```diff
@@ -548,6 +553,7 @@ class PlansSetup extends React.Component {
 
 		return (
 			<div className="jetpack-plugins-setup">
+				{ this.renderPageViewTracker() }
```
Comment (gwwar) [inline client/my-sites/plugins/jetpack-plugins-setup/index.jsx]: The function isn't doing anything interesting. Let's inline the component call here.

**Comment by gwwar** [comment]
Almost! This is good to 🚢 after behavior in `plugins/manage/:site` is tidied.

**Comment by Copons** [comment]
I'm not getting page views for:

- /plugins/active/:site
- /plugins/inactive/:site
- /plugins/updates/:site
- /plugins/upload/:site
- /plugins/:plugin/eligibility/:site

Also can confirm that:
`/plugins/manage/:site` records as `/plugins/:site` (which is a different view).

File: client/my-sites/plugins/jetpack-plugins-setup/index.jsx (hunk 1 above)
Comment (vindl) [reply client/my-sites/plugins/jetpack-plugins-setup/index.jsx]: Updated in 152380974e1fe5e4edc33de55a46102262327ecb.

**Comment by vindl** [comment]
> plugins/manage/:site x second page recording of plugins/:site for Jetpack sites

Resolved this in 152380974e1fe5e4edc33de55a46102262327ecb.

**Comment by kwight** [comment]
I just ran through this; `plugins/manage/:site` is ✅, I had the same missing results as @Copons .

**Comment by vindl** [comment]
> I'm not getting page views for:
/plugins/active/:site
/plugins/inactive/:site
/plugins/updates/:site
/plugins/upload/:site
/plugins/:plugin/eligibility/:site

@Copons @kwight I just pushed an update that should add all of the missing stats for the above paths - 07ae10b02739a34c2c170dc05961739b9a6fd2ca.

**Pull request merged**